lib/DxilPIXPasses/PixPassHelpers.cpp

```diff
@@ -41,8 +41,16 @@ bool IsAllocateRayQueryInstruction(llvm::Value const *Val) {
   if (Val != nullptr) {
     if (llvm::Instruction const *Inst =
             llvm::dyn_cast<llvm::Instruction>(Val)) {
-      return hlsl::OP::IsDxilOpFuncCallInst(Inst,
-                                            hlsl::OP::OpCode::AllocateRayQuery);
+      if (hlsl::OP::IsDxilOpFuncCallInst(Inst,
+                                         hlsl::OP::OpCode::AllocateRayQuery))
+        return true;
+      if (auto PHI = llvm::dyn_cast<llvm::PHINode>(Inst)) {
```

Comment:
Incoming value for PHI might be another PHI, so I think this might need to be recursive?
It may be possible to see a select instead of PHI as well.

So, I think a robust solution requires a little more work.

Also, the name `IsAllocateRayQueryInstruction` seems like it should be something more like `IsRayQueryHandle`.

Reply:
For testing, you might need an IR test to get a select scenario if it's hard to produce in HLSL.

Reply:
Woah, didn't know select existed! Thanks!
Also, it seems like phis can recursively refer to each other, which breaks my brain.

Reply:
Addressed the recursive phi issue and added select checking. switch() with -O3 exercises it.